### PR TITLE
[plugin.video.catchuptvandmore] 0.2.21

### PR DESCRIPTION
**Pull request by SylvainCecchetto**
### Description

[Version 0.2.21]
[Fix/Add] Catch-up TV - Japaneses channels (JP)
[Add] Catch-up TV - Télé Grenoble (FR)
[Fix] Catch-up TV - Arte (WO)
[Fix] Add check for download mode
[Fix] Disable reuselanguageinvoker to fix Youtube-DL download feature
[Fix] Catch-up TV - La 1ère (FR)
[Other] Include Codequick framework
[Fix] Live TV - RTLPLAY BE (BE)

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.catchuptvandmore/addon.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
   <extension point="xbmc.addon.metadata">
     <!-- Set reuselanguageinvoker to false while developing the addon, making it only true for the production version -->
     <!-- See https://kodi.wiki/view/Addon.xml#.3Creuselanguageinvoker.3E -->
-    <reuselanguageinvoker>true</reuselanguageinvoker>
+    <!-- <reuselanguageinvoker>true</reuselanguageinvoker> -->
```
Comment (enen92) [inline plugin.video.catchuptvandmore/addon.xml:24]: intentional @SylvainCecchetto ?

Comment (SylvainCecchetto) [reply plugin.video.catchuptvandmore/addon.xml:24]: Yes, because of this https://github.com/ruuk/script.module.youtube.dl/pull/56

**Comment by SylvainCecchetto** [comment]
Hi enen, do you need something more for this PR? 🙂

**Comment by enen92** [comment]
I'll be honest, I don't like the codequick change as it violates the usage of script.modules. If the original author is not available (will he comeback at some point or has he just given up?) the logical action in opensource world is just to fork his work and rebrand (checking the license). If everyone does what you guys are doing here, it's simply not manageable on the long run. We are reviewing modules and reviewing them patched (with the same name) as part of an addon again. 

I am fine with merging this and allowing the lib **temporarily** but I really would prefer another option at some point. A complete rebrand provided as a script.module to the outside world, or an internal rebrand if it is to be used just by this plugin.
You are a team member so I think you'll take this into consideration and try to do things properly for the benefit of everyone else.

**Comment by SylvainCecchetto** [comment]
Thank you for the time you took to explain all those things.

We will try to contact willford, the maintainer of CodeQuick to know if he wants to keep maintaining its framework in the future or not.

If you are still agree, we are ok to merge this PR with the embedded framework. That gave us a little time to think about 0.2.22 and what to do with CodeQuick.

Thank you.

**Comment by enen92** [comment]
Take it in consideration for a next update either way if possible. The addon is already big enough :)

**Pull request merged**